packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2679,44 +2678,73 @@ export class ContainerRuntime
 				inboundBatch,
 				local,
 			);
+
 			if (messagesWithPendingState.length > 0) {
-				messagesWithPendingState.forEach(({ message, localOpMetadata }) => {
-					const msg: MessageWithContext = {
-						message,
-						local,
-						isRuntimeMessage: true,
-						savedOp,
-						localOpMetadata,
-					};
-					this.ensureNoDataModelChanges(() => this.processRuntimeMessage(msg));
-				});
+				this.processBatch(
+					messagesWithPendingState,
+					local,
+					savedOp,
+					true /* modernRuntimeMessage */,
+				);
 			} else {
 				this.ensureNoDataModelChanges(() => this.processEmptyBatch(inboundBatch, local));
 			}
 		} else {
-			// Check if message.type is one of values in ContainerMessageType
-			// eslint-disable-next-line import/no-deprecated
-			if (isRuntimeMessage(messageCopy)) {
-				// Legacy op received
-				this.ensureNoDataModelChanges(() =>
-					this.processRuntimeMessage({
-						message: messageCopy as InboundSequencedContainerRuntimeMessage,
-						local,
-						isRuntimeMessage: true,
-						savedOp,
-					}),
-				);
-			} else {
-				// A non container runtime message (like other system ops - join, ack, leave, nack etc.)
-				this.ensureNoDataModelChanges(() =>
-					this.observeNonRuntimeMessage({
-						message: messageCopy as InboundSequencedNonContainerRuntimeMessage,
-						local,
-						isRuntimeMessage: false,
-						savedOp,
-					}),
-				);
-			}
+			const batch = [{ message: messageCopy, localOpMetadata: undefined }];
+			this.processBatch(batch, local, savedOp, undefined /* modernRuntimeMessage */);
```

Comment:
Am I correct, that there is observable change here - we will raise beginBatch & endBatch events for system messages (where we did not in the past)? Is this desired?